.pipelines/cni/cilium/nightly-release-test.yml

```diff
@@ -40,19 +54,54 @@ stages:
             inputs:
               containerRegistry: $(CONTAINER_REGISTRY)
               command: "logout"
+
   - stage: cilium_nightly
     displayName: E2E - Cilium Nightly
+    dependsOn:
+      - init
+      - setup
     variables:
       GOPATH: "$(Agent.TempDirectory)/go" # Go workspace path
       GOBIN: "$(GOPATH)/bin" # Go binaries path
       modulePath: "$(GOPATH)/src/github.com/Azure/azure-container-networking"
+      commitID: $[ stagedependencies.setup.env.outputs['EnvironmentalVariables.commitID'] ]
     jobs:
+      - template: ../../templates/create-cluster.yaml
+        parameters:
+          name: cilium_overlay_nightly
+          displayName: Cilium on AKS Overlay
+          clusterType: cilium-overlay-up
+          clusterName: ciliumnightly-$(commitID)
+          vmSize: Standard_B2ms
+          k8sVersion: ""
+          dependsOn: ""
+          region: $(LOCATION)
       - job: cilium_nightly
+        dependsOn:
+          - cilium_overlay_nightly
         pool:
           name: $(BUILD_POOL_NAME_DEFAULT)
         steps:
         - template: ../../singletenancy/cilium-overlay/cilium-overlay-e2e-step-template.yaml
           parameters:
             name: "cilium_nightly"
             testDropgz: ""
-            clusterName: "ciliumnightly"
+            clusterName: ciliumnightly-$(commitID)
+      - job: delete
```

Comment:
We should add a condition for deleting the cluster `always` . or is it a part of the template ?

Reply:
I'll add it back. Must have missed it. Good find!